src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +30,139 @@ public Task Get()
+                var topology = ServerStore.Cluster.ReadDatabaseTopology(context, database);
+                nodes = GetResponsibleNodes(topology, databaseGroupId, pullReplication);
+            }
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var output = new DynamicJsonArray();
+                var clusterTopology = ServerStore.GetClusterTopology();
+                foreach (var node in nodes)
+                {
+                    output.Add(clusterTopology.GetUrlFromTag(node));
+                }
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["Results"] = output
+                });
+            }
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/info/remote-task/tcp", "GET", AuthorizationStatus.RestrictedAccess)]
+        public Task GetRemoteTaskTcp()
+        {
+            var remoteTask = GetStringQueryString("remote-task");
+            var database = GetStringQueryString("database");
+
+            Authenticate(HttpContext, ServerStore, database, remoteTask);
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var output = Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl());
+                context.Write(writer, output);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static void Authenticate(HttpContext httpContext, ServerStore serverStore, string database, string remoteTask)
+        {
+            var feature = httpContext.Features.Get<IHttpAuthenticationFeature>() as RavenServer.AuthenticateConnection;
+
+            if (feature == null) // we are not using HTTPS 
+                return;
+
+            switch (feature.Status)
+            {
+                case RavenServer.AuthenticationStatus.Allowed:
+                case RavenServer.AuthenticationStatus.Operator:
+                case RavenServer.AuthenticationStatus.ClusterAdmin:
+                    // we can trust this certificate
+                    return;
+
+                case RavenServer.AuthenticationStatus.UnfamiliarCertificate:
+                    using (serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                    using (context.OpenReadTransaction())
+                    {
+                        var pullReplication = serverStore.Cluster.ReadPullReplicationDefinition(database, remoteTask, context);
+                        var cert = httpContext.Connection.ClientCertificate;
+                        if (pullReplication.CanAccess(cert?.Thumbprint, out var err) == false)
```

Comment:
what happens if the db doesn't exists? What happens if the pull replication doesn't exists?

Reply:
If the db doesn't exists `throw new DatabaseDoesNotExistException($"The database '{database}' doesn't exists.")`
If the pull replication doesn't exists `throw new InvalidOperationException($"Pull replication with the name '{definitionName}' isn't defined for the database '{database}'.");`